server/services/store/sqlstore/migrate.go

```diff
@@ -213,6 +225,194 @@ func (s *SQLStore) Migrate() error {
 	return engine.ApplyAll()
 }
 
+func (s *SQLStore) migrateSchemaVersion(migrations []*models.Migration) error {
+	migrationNeeded, err := s.isSchemaMigrationNeeded()
+	if err != nil {
+		return err
+	}
+
+	if !migrationNeeded {
+		return nil
+	}
+
+	s.logger.Info("Migrating schema migration to new format")
+
+	legacySchemaVersion, err := s.getLegacySchemaVersion()
+	if err != nil {
+		return err
+	}
+
+	if err := s.createTempSchemaTable(); err != nil {
+		return err
+	}
+
+	if err := s.populateTempSchemaTable(migrations, legacySchemaVersion); err != nil {
+		return err
+	}
+
+	if err := s.useNewSchemaTable(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *SQLStore) isSchemaMigrationNeeded() (bool, error) {
+	// Check if `dirty` column exists on schema version table.
+	// This column exists only for the old schema version table.
+
+	// SQLite needs a bit of a special handling
+	if s.dbType == model.SqliteDBType {
+		return s.isSchemaMigrationNeededSQLite()
+	}
+
+	query := s.getQueryBuilder(s.db).
+		Select("count(*)").
+		From("information_schema.COLUMNS").
+		Where(sq.Eq{
+			"TABLE_NAME":  s.tablePrefix + "schema_migrations",
+			"COLUMN_NAME": "dirty",
+		})
+
+	row := query.QueryRow()
+
+	var count int
+	if err := row.Scan(&count); err != nil {
+		s.logger.Error("failed to check for columns of schema_migrations table", mlog.Err(err))
+		return false, err
+	}
+
+	return count == 1, nil
+}
+
+func (s *SQLStore) isSchemaMigrationNeededSQLite() (bool, error) {
+	// the way to check presence of a column is different
+	// for SQLite. Hence, the separate function
+
+	query := fmt.Sprintf("PRAGMA table_info(\"%sschema_migrations\");", s.tablePrefix)
+	rows, err := s.db.Query(query)
+	if err != nil {
+		s.logger.Error("SQLite - failed to check for columns in schema_migrations table", mlog.Err(err))
+		return false, err
+	}
+
+	defer s.CloseRows(rows)
+
+	data := [][]*string{}
+	for rows.Next() {
+		// PRAGMA returns 6 columns
+		row := make([]*string, 6)
+
+		err := rows.Scan(
+			&row[0],
+			&row[1],
+			&row[2],
+			&row[3],
+			&row[4],
+			&row[5],
+		)
+		if err != nil {
+			s.logger.Error("error scanning rows from SQLite schema_migrations table definition", mlog.Err(err))
+			return false, err
+		}
+
+		data = append(data, row)
+	}
+
+	nameColumnFound := false
+	for _, row := range data {
+		if len(row) >= 2 && *row[1] == "dirty" {
+			nameColumnFound = true
+			break
+		}
+	}
+
+	return nameColumnFound, nil
+}
+
+func (s *SQLStore) getLegacySchemaVersion() (uint32, error) {
+	query := s.getQueryBuilder(s.db).
+		Select("version").
+		From(s.tablePrefix + "schema_migrations")
```

Comment:
Maybe we should just store `s.tablePrefix + "schema_migrations"` value somewhere. It's being used many times.

Reply:
We don't do that anywhere in Focalboard.